packages/playground/app/assets/js/playground.js

```diff
@@ -0,0 +1,232 @@
+      const urlParameters = getParams(window.location.href);
+
+      // Casts to array because of IE11
+
+      // Set selected option of contentDensitySwitch
+      Array.prototype.slice.call(themeSwitch.querySelectorAll("ui5-option")).forEach(function(option) {
+        option.selected = option.value === urlParameters["sap-ui-theme"];
+      });
+
+      // Set selected option of themeSwitch
+      Array.prototype.slice.call(contentDensitySwitch.querySelectorAll("ui5-option")).forEach(function(option) {
+        if (urlParameters["sap-ui-compactSize"] === "true") {
+          option.selected = option.textContent === "Compact";
+        } else {
+          option.selected = option.textContent === "Cozy";
+        }
+      });
+
+       // Set selected option of themeSwitch
+       Array.prototype.slice.call(textDirectionSwitch.querySelectorAll("ui5-option")).forEach(function(option) {
+        if (urlParameters["sap-ui-rtl"] === "true") {
+          option.selected = option.textContent === "RTL";
+        } else {
+          option.selected = option.textContent === "LTR";
+        }
+      });
+
+      dialog.open();
+    });
+
+    cancelButton.addEventListener("click", function(event) {
+      dialog.close();
+    });
+
+    applyButton.addEventListener("click", function(event) {
+      const theme = themeSwitch.selectedOption.value,
+        contentDensity = contentDensitySwitch.selectedOption.textContent,
+        textDirection = textDirectionSwitch.selectedOption.textContent;
+
+        // Not implemented with string literals, beacause of IE11
+        let newLocation = location.origin + location.pathname + "?sap-ui-theme=";
+        newLocation += theme;
+        newLocation += "&sap-ui-compactSize=";
+        newLocation += contentDensity === "Compact";
+        newLocation +=  "&sap-ui-rtl=";
+        newLocation += textDirection === "RTL";
+
+        window.location = newLocation;
+      
+      dialog.close();
+    });
+  }
+}
+
+function setTheme() {
+  const currentTheme = getParams(window.location.href)["sap-ui-theme"];
+  if (currentTheme === "sap_belize_hcb") {
+    document.body.classList.add("hcb");
+  } else {
+    document.body.classList.remove("hcb");
+  }
+}
+
+function getParams(url) {
+  const params = {};
+  const parser = document.createElement('a');
+  parser.href = url;
+  const query = parser.search.substring(1);
+  const vars = query.split('&');
+  for (let i = 0; i < vars.length; i++) {
+    const pair = vars[i].split('=');
+    params[pair[0]] = decodeURIComponent(pair[1]);
+  }
+  return params;
+}
+
+// Site search
+function initSearch() {
+    var index = lunr(function () {
+      this.ref('id');
+      this.field('title', { boost: 20 });
+      this.field('content', { boost: 10 });
+      this.field('url');
+    });
+  
+    // Get the generated search_data.json file so lunr.js can search it locally.
+  
+    sc = document.getElementsByTagName("script");
```

Comment:
Not defined variables: sc, source, idx, s, jsPath, jsonPath and I stop looking for more. Just add "use strict" at the top to find them all.

Reply:
This code comes from the theme, but I fixed it.